plugins/mcreator-core/blockly/js/field_data_list_selector.js

```diff
@@ -93,48 +71,32 @@ class FieldDataListSelector extends Blockly.FieldLabelSerializable {
         }
     };
 
-    // We store only the actual value in the text content, the readable name is loaded with the procedure
-    toXml(fieldElement) {
-        fieldElement.textContent = this.getValue();
-        return fieldElement;
-    };
-
-    // We load the readable name again after opening the procedure, in case the entry has a new readable name
-    fromXml(fieldElement) {
-        if (fieldElement && fieldElement.textContent) {
-            let readableName = javabridge.getReadableNameOf(fieldElement.textContent, this.type);
-            if (!readableName) // The readable name is an empty string because it couldn't be found
-                readableName = fieldElement.textContent; // In this case, we use the actual value
-            this.entry = fieldElement.textContent + ',' + readableName;
-        } else
-            this.entry = FieldDataListSelector.getDefaultEntry();
-        this.updateDisplay();
+    // Update the value of this selector
+    doValueUpdate_(newValue) {
+        if (newValue !== this.value_) { // If the value is different, update the readable name
+            this.updateReadableName(newValue);
+        }
+        super.doValueUpdate_(newValue);
     };
 
-    // Returns the readable text
+    // Get the text that is shown in the Blockly editor
     getText_() {
-        if (this.entry && this.entry.split(',').length === 2) {
-            return this.entry.split(',')[1];
-        }
-        return javabridge.t('blockly.extension.data_list_selector.no_entry');
-    }
+        return this.readableName || FieldDataListSelector.getDefaultText();
+    };
 
-    // Returns the actual value of the selected entry. Only this value is saved in the procedure XML
-    getValue() {
-        if (this.entry && this.entry.split(',').length === 2) {
-            return this.entry.split(',')[0];
+    // Update the readable name of this field
+    updateReadableName(value) {
+        // First check if there's a cached readable name (after selecting a value)
+        if (this.cachedReadableName) {
+            this.readableName = this.cachedReadableName;
+            this.cachedReadableName = null

```

Comment:
Shouldn't there be a semicolon at the end?

Reply:
Good catch